porch/test/e2e/e2e_test.go

```diff
@@ -703,7 +704,19 @@ func (t *PorchSuite) TestDeleteDraft(ctx context.Context) {
 	var draft porchapi.PackageRevision
 	t.mustExist(ctx, client.ObjectKey{Namespace: t.namespace, Name: created.Name}, &draft)
 
-	// Delete the package
+	// Try to delete the package. This should fail because it hasn't been proposed for deletion.
+	t.DeleteL(ctx, &porchapi.PackageRevision{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: t.namespace,
+			Name:      created.Name,
+		},
+	})
+	t.mustExist(ctx, client.ObjectKey{Namespace: t.namespace, Name: created.Name}, &draft)
+
+	// Propose deletion and then delete the package
+	draft.Spec.DeletionProposed = true
+	t.UpdateF(ctx, &draft)
```

Comment:
The test framework has code that cleans up resources in the cluster after the test is finished. We might want to update that too so it correctly propose deletion and then deletes revisions.

Reply:
I can take a look at some point, but with only requiring [published packages to need this deletion approval](https://github.com/GoogleContainerTools/kpt/pull/3678#discussion_r1048862135), I think this becomes much less of an issue.